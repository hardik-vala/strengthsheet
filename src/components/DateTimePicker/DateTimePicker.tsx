import RNCDateTimePicker from "@react-native-community/datetimepicker";
import { View } from "react-native";
import { Text } from "react-native-paper";

interface DateTimePickerProps {
  label: string;
  testID: string;
  value: Date;
  isDate: boolean;
  onChange: (e, d: Date) => void;
}

export function DateTimePicker({
  label,
  testID,
  value,
  isDate,
  onChange,
}: DateTimePickerProps) {
  return (
    <View
      style={{
        alignItems: "center",
        flexDirection: "column",
        padding: 5
      }}
    >
      <Text variant="bodyLarge">{label}</Text>
      <RNCDateTimePicker
        testID={testID}
        value={value}
        mode={isDate ? "date" : "time"}
        is24Hour={true}
        onChange={onChange}
        style={{marginTop: 3}}
      />
    </View>
  );
}
