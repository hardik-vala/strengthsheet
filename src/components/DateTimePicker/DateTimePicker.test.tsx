import { render } from "@testing-library/react-native";
import React from "react";

import { DateTimePicker } from "./DateTimePicker";

describe("DateTimePicker", () => {
  it("renders correctly", () => {
    const tree = render(
      <DateTimePicker
        label="My picker"
        testID="pickerTestId"
        value={new Date("2023-08-27")}
        isDate={true}
        onChange={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
