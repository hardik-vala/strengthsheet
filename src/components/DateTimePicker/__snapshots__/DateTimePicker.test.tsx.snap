// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "padding": 5,
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    My picker
  </Text>
  <RNDateTimePicker
    date={1693094400000}
    displayIOS="default"
    enabled={true}
    is24Hour={true}
    mode="date"
    onChange={[Function]}
    onPickerDismiss={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginTop": 3,
      }
    }
    testID="pickerTestId"
  />
</View>
`;
